docs/README.md

```diff
@@ -0,0 +1,148 @@
+
+#### Pinned Rollout
+For more cautious and manual workflows, users may follow the Pinned Release pattern. It is executed using the following:
+
+* Start with a KFService that has a "stable" or "golden" default definition.
+* Modify the KFService to set the new ModelSpec under the `canary`.
+* Set trafficPercent to 0
+* Observe the KFService until the new revision progresses
+    * Note: traffic is still sent to "default"
+* Execute any inspection or testing against `canary`
+    * Note: The revision will be available under a specialized URI (e.g. revisionName.name.namespace.cluster.internal)
+* Once complete, change the key `canary` to `default` and delete the original `default` section.
+* The underlying resource will swap traffic which can be observed in `status`
+
+#### Canary Rollout
+For the most advanced deployments, users may follow the Canary Rollout Pattern. It is executed using the following:
+
+* Start with a KFService that has a "stable" or "golden" default definition.
+* Modify the KFService to set the new ModelSpec under the `canary`.
+* Set trafficPercent to some integer value (e.g. 10)
+* Observe the KFService until the new revision progresses from "latestCreatedRevisionName" to "latestReadyRevisionName"
+    * Note: `100 - trafficPercent` is still sent to "default"
+* Continue incremented trafficPercent in accordance with your desired canarying strategy
+* Once complete, change the key `canary` to `default` and delete the original `default` section.
+* The underlying resource will swap traffic which can be observed in `status`
+
+#### Rollback
+One of the most compelling characteristics of "infrastructure-as-code" interfaces like the Kubernetes Resource Model is the ability to revert a configuration to a last known golden state. The default/canary pattern defines the entire current state, enabling GitOps rollback.
+
+* revert to previously safe commit
+* kubectl apply KFService.yaml
+
+It's possible to revert to a commit the contains `canary` in the spec. The state will be reconciled to rollback to a configuration that runs both `default` and `canary`. This is possible with the data model, but somewhat of an anti-pattern for safe rollbacks. We recommend reverting to KFService spec that contains only `default`.
+
+### Resources
+The default implementation will attempt to provide sane defaults for CPU and MEMORY. If more control is needed, we defer to [Specifying Resources in Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container).
+
+By default, a Service will be configured with requests and limits of:
+* 1 CPU
+* 2 GB MEM
+
+In addition to specifying nvidia/gpu count, users may target specific GPUs on specific platforms by using annotations (e.g. "cloud.google.com/gke-accelerator": "nvidia-tesla-t4"):
+
+### Replicas
```

Comment:
Do we want to document the auto scale down to 0 case which is default?

Reply:
Yes. I'll add a note on this.